src/hardware/voodoo.cpp

```diff
@@ -2316,8 +2317,9 @@ do																				\
 				depthval = 0xffff;												\
 			else																\
 			{																	\
-				int exp = count_leading_zeros(temp);							\
-				depthval = ((exp << 12) | ((~temp >> (19 - exp)) & 0xfff));		\
+				const auto exp = count_leading_zeros(temp);                     \
+				const auto right_shift = std::max(0, 19 - exp);	                \
+				depthval = ((exp << 12) | ((~temp >> right_shift) & 0xfff));    \
```

Comment:
Does this fix an observable, obvious rendering bug? We can't make any assumptions on how the Voodoo hardware works, this might be intentional and exact recreation of the behaviour of the real thing.

Reply:
It expects to shift the `~temp` value to the right by up to 19 bits (at the most).

At the least, we can right-shift by zero places (and not change anything).

However if the `temp` value has more than 19 leading zero, then the (original) code actually tries to right-shift `~temp` by /negative/ bits.  For example: `~temp >> -3` : that's garbage land and triggers a UBSAN crash.

So I'm just stopping it from driving into negative territory and triggering undefined behavior.

But in terms of valid-shift ranges, that's left intact. The shift range from 19 at the most all the way down to 0. 

(If the code needs to trigger UBSAN errors to work correctly, then it's not passing CI and it goes back into the expeterminal-flag jail :-)!)

Reply:
Thanks for the explanation. I'm actually not going into great detail with the review, just flagging things that go beyond simple refactoring and look suspcious 😄 I can stop that if it's bothersome and you've done extensive testing.

Reply:
Please keep flagging!
I did test each commit as I went - but there's always a chance of typos and my own mistakes! Need all the help I can get ❤️👍